File: BaekJoon/solve/2309/Combination.cpp
#include <iostream>
#include <vector>

using namespace std;

const vector<int> nums = {1, 2, 3, 4, 5, 6, 7, 8, 9};
int cnt = 0;

void make_combination(int index , vector<int> current_nums) { 
  if(current_nums.size() == 3) {
    for(int num : current_nums) {
      cout << num << " ";
    }
    cout << '\n';
    return;
  }
  for(int i = index; i < nums.size(); ++i) { 
    vector<int> next_nums = current_nums;
    next_nums.push_back(nums[i]);
    make_combination(i + 1, next_nums);
  }
}

int main(void) {
  make_combination(0, { });
}

File: BaekJoon/solve/2309/Combination.swift
let arr = [1, 2, 3, 4, 5]

func makeCombination(depth: Int, index: Int, currentArray: [Int]) {
  if depth == 3 { 
    print(currentArray)
    return 
  }

  for i in index..<arr.count { 
    makeCombination(depth: depth + 1, index: i + 1, currentArray: currentArray + [arr[i]])
  }
}

makeCombination(depth: 0, index: 0, currentArray: [])
